src/util/gc_byte.rs

```diff
@@ -1,11 +1,35 @@
-use crate::util::ObjectReference;
+use crate::util::side_metadata::{SideMetadata, SideMetadataID};
+use crate::util::{constants, ObjectReference};
 use crate::vm::ObjectModel;
 use crate::vm::VMBinding;
 use std::sync::atomic::{AtomicU8, Ordering};
 
+use super::Address;
+
 /// This struct encapsulates operations on the per-object GC byte (metadata)
 pub struct GCByte {}
 
+// For performance reasons, mutable static is used over lazy_static or sync::once.
+// This requires several unsafe blocks in the current file, but it looks safe to use these unsafe blocks, because we only assign the mutable static once.
+//
+// NOTE: A more suitable implementation may also be possible.
+static mut SIDE_GCBYTE_ID: SideMetadataID = SideMetadataID::new();
+
+#[allow(clippy::cast_ref_to_mut)]
+#[allow(clippy::mut_from_ref)]
+pub(crate) unsafe fn init_side_gcbyte() {
+    let res = SideMetadata::request_meta_bits(
+        // constants::BITS_IN_BYTE,
+        2,
+        constants::LOG_BYTES_IN_WORD as usize,
```

Comment:
The minimum object size in MMTk is one word (two words in some cases).
I don't understand why the minimum alignment is less than that.

Anyway, for side metadata, the minimum object size is the key factor.
Maybe I should change the `log_bytes_align` to e.g. `min_data_size` to make it more clear?

Reply:
Yeah, that sounds fine. We have a min object size defined somewhere when you added the assertion to check min object size.

Reply:
We check the minimum allocation size by:
```
crate::util::forwarding_word::check_alloc_size::<VM>(size);
```
which calculates the minimum allocation size based-on `VMObjectModel::HAS_GC_BYTE` and `VMObjectModel::GC_BYTE_OFFSET`.

Now, I'm thinking what if the client VM has a bigger min allocation size? To avoid waste of metadata space, will it be useful if we allow them to specify their own min alloc size?

Reply:
I think that's reasonable. But we don't need to include this in this PR. My point was that if we assume min object size or min alignment, we should use the same constant rather than defining it several times.

Reply:
Then, I think I should open an issue for it and assign it to myself, otherwise I will certainly forget it 😄